### DISCUSSION TITLE
gatsby-plugin-theme-ui - Add compatibility for Gatsby 3

### DISCUSSION THREAD

**Discussion by moyarich** [system-ui/theme-ui#1558] [Ideas]
Gatsby is now at [version 3,](https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#for-plugin-maintainers) this package is using Gatsby 2 as a peerDependency. 

When can we expect a [version bump](https://github.com/semver/semver/issues/502) for the [gatsby-plugin-theme-ui ](https://github.com/system-ui/theme-ui/tree/develop/packages/gatsby-plugin-theme-ui)to its next major version ?

**Comment by lachlanjc** [comment]
Great idea! We'd love a PR. Is it working right now but with a warning, or are there changes to the code we need?

**Reply by moyarich** [reply]
I don't know enough about the plugin to test all functionality myself but here is how to trigger the issue.

# recreate dependency issues:

**Could not resolve _react_ peer dependency issue :**
*react@^16.11.0 **from** theme-ui@0.3.5*

```
mkdir testapp && cd testapp && npm init -y
npm install gatsby react react-dom
npm install theme-ui
```

**Could not resolve _gatsby_ peer dependency issue :**
*gatsby@^2.13.1 **from** gatsby-plugin-theme-ui@0.3.5*
```
npm install gatsby-plugin-theme-ui 
```

**Reply by lachlanjc** [reply]
Ah yeah. We’ve got issue #1563 for tracking this fix!
